source/app/models/url.rb

```diff
@@ -0,0 +1,53 @@
+class Url < ActiveRecord::Base
+  require 'net/http'
+  
+  before_save :generate_shortened_url
+
+  validates :address, presence: true
+  validates :address, format: { with: /\Ahttps?:\/\/.*/, message: "must begin with 'http://' or 'https://'" }
+  validates :address, uniqueness: true
+  validate :url_is_correct_format, :url_responds_to_http_request
+
+  def url_is_correct_format
+  	unless address =~ /\A#{URI.regexp}\z/
+  	  errors.add(:address, "is not a valid url format")
+  	end
+  end
+
+  def url_responds_to_http_request
+  	url = URI.parse(address)
```

Comment:
Could a bad actor use this logic to DOS another site possibly?

Reply:
Yeah, they could. Are there any specific safe practices I could look up? Most of my generic searches just send me to security companies.